spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/util/LocalAdressResolver.java

```diff
@@ -0,0 +1,129 @@
+package org.springframework.cloud.sleuth.util;
+
+import java.net.*;
+import java.nio.ByteBuffer;
+import java.util.Enumeration;
+
+import org.apache.commons.logging.Log;
+
+/**
+ * Utility class for getting local Ip address from network interfaces.
+ * Picks up first non loopback and up address.
```

Comment:
this code is similar to a lot of things, but it is substantial enough to warrant a question. Was this inspired by another project's code? if so, note the class it was derived from as it usually helps explain choices.